diff --git a/tensorflow/core/util/mkl_util.h b/tensorflow/core/util/mkl_util.h
index 34db96075d..fb2cb0913c 100644
--- a/tensorflow/core/util/mkl_util.h
+++ b/tensorflow/core/util/mkl_util.h
@@ -1578,13 +1578,6 @@ class MklDnnData {
     return user_memory_->get_data_handle();
   }
 
-  /// Set function for data buffer of user memory primitive.
-  inline void* SetUsrMemDataHandle(void* data_buffer) {
-    CHECK_NOTNULL(user_memory_);
-    CHECK_NOTNULL(data_buffer);
-    return user_memory_->set_data_handle(data_buffer);
-  }
-
   /// Set function for data buffer of user memory primitive.
   inline void SetUsrMemDataHandle(const Tensor* tensor) {
     CHECK_NOTNULL(user_memory_);
